rules-tests/TypeDeclaration/Rector/FunctionLike/ReturnTypeDeclarationRector/FixtureForPhp81/do_not_crash_on_numeric_string.php.inc

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\FunctionLike\ReturnTypeDeclarationRector\FixtureForPhp81;
+
+class DoNotCrashOnNumericString
+{
+    public function run(): string
+    {
+        return bcadd('10', '5');
+    }
+}
```

Comment:
Could you add test fixture withotu `): string` return type and the other half that adds it? So we're sure this type is completed.

```php
<?php

namespace Rector\Tests\TypeDeclaration\Rector\FunctionLike\ReturnTypeDeclarationRector\FixtureForPhp81;

class DoNotCrashOnNumericString
{
    public function run()
    {
        return bcadd('10', '5');
    }
} 

?>
-----
<?php

namespace Rector\Tests\TypeDeclaration\Rector\FunctionLike\ReturnTypeDeclarationRector\FixtureForPhp81;

class DoNotCrashOnNumericString
{
    public function run(): string
    {
        return bcadd('10', '5');
    }
} 

?>
```

Reply:
New PHPStan 1.3 just released https://github.com/phpstan/phpstan/releases/tag/1.3.0 and as we use `^1.2` in composer.json, it will automatically updated and https://github.com/rectorphp/rector-src/pull/1589 and make error as new PR https://github.com/rectorphp/rector-src/pull/1589

I will pin to PHPStan 1.2 and add new fixture for it.

Reply:
> I will pin to PHPStan 1.2 and add new fixture for it.

Thanks :+1:

Reply:
@TomasVotruba I applied:

- temporary pin phpstan to 1.2 as 1.3 cause error like in PR https://github.com/rectorphp/rector-src/pull/1589 as currently we use `^1.2`

- add new fixture for no return declaration, it seems currently cause different result diff:

```diff
-    public function run(): string
+    public function run()
```

Reply:
oh, it seems because `bcadd()` function not exists, so no return type is expected.

Reply:
Fixed with add the missing function and add string handling https://github.com/rectorphp/rector-src/pull/1588/commits/50a27aa601de9cb26629cfa14754b932d476ed27 so it can add return type declaration.

Reply:
Thank you :+1: